### PR TITLE
chore: Includes GH actions dependency review in PR checks

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

We have this action in our [CFN repository](https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/master/.github/workflows/code-health.yaml#L51) but was missing here.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Review by marcosuma — APPROVED** [review]

**Comment by AgustinBettati** [comment]
Will wait to merge as we saw a false positive raised in CFN repository [PR](https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/889), raised an issue to the dependency-review-action to get more clarity: https://github.com/actions/dependency-review-action/issues/676.

File: .github/workflows/code-health.yml [hunk 1]
```diff
@@ -70,6 +70,14 @@ jobs:
           go-version-file: 'go.mod'
       - name: website lint
         run: make tools && make website-lint
+  dependency-review:
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'pull_request' }}
```
Comment (lantoli) [inline .github/workflows/code-health.yml:75]:
```suggestion
   
```

I would remove the check for pr and will do always, e.g. why not do it if workflow_dispatch?

**Review by lantoli — APPROVED** [review]
thx!   is it in cdk repo as well?

**Comment by AgustinBettati** [comment]
Closing this PR for now, https://github.com/actions/dependency-review-action/issues/676 confirmed that this action is not able to infer the proper ordering of pinned GitHub Actions versions so this will likely bring up false positives.

**Pull request closed without merging**